lib/date_range.rb

```diff
@@ -0,0 +1,28 @@
+module Hotel
+  class DateRange
+    attr_accessor :start_date, :end_date
+
+    def initialize(start_date, end_date)
+      # if start date is on or after end Date, raise an exception
+      if start_date >= end_date
+        raise ArgumentError, " Invalid Date Range."
+      
+      end
+      
+      @start_date = start_date
+      @end_date = end_date
+    end
+    # check if date selected overlaps with the date_range
+    def overlap?(new_start_date, new_end_date)
```

Comment:
Nice use of a boolean expression to return `true` or `false`. You don't even need a conditional :)